include/CL/sycl/vec/detail/vec.hpp

```diff
@@ -90,8 +94,47 @@ class vec : public detail::small_array<DataType,
     return std::tuple_cat(flatten<V>(i)...);
   }
 
+protected:
+  __swizzled_base_vec__<DataType, 1>swizzle(int first) const {
+    return (*this)[first];
+  }
+
+  template<typename T, typename... Ts>
+  __swizzled_base_vec__<DataType, sizeof...(Ts) + 1>swizzle(T first, Ts... swizzleIndexes) const {
+    return __swizzled_base_vec__<DataType, sizeof...(Ts) + 1> {swizzle(first), swizzle(swizzleIndexes...) };
+  }
+
```

Comment:
You can replace both functions by
```  
/// Apply a swizzle operation from swizzleIndexes parameters
  template<typename... Ts>
  __swizzled_base_vec__<DataType, sizeof...(Ts)>
  swizzle(Ts... swizzleIndexes) const {
    // Construct a new vector from an elemental swizzle of each element
    return __swizzled_base_vec__<DataType, sizeof...(Ts)> {
      (*this)[swizzleIndexes]... };
  }
```
which works for me without recursion.
Modern C++ is simpler to write and should compile faster. :-)

Reply:
You can fix my pseudo-English comments.